js/eventloop/eventloop_test.go

```diff
@@ -143,3 +144,72 @@ func TestEventLoopPanicOnDoubleCallback(t *testing.T) {
 	require.Less(t, time.Second, took)
 	require.Greater(t, time.Second+time.Millisecond*100, took)
 }
+
+func TestEventLoopRejectUndefined(t *testing.T) {
+	t.Parallel()
+	vu := &modulestest.VU{RuntimeField: goja.New()}
+	loop := eventloop.New(vu)
+	err := loop.Start(func() error {
+		_, err := vu.Runtime().RunString("Promise.reject()")
+		return err
+	})
+	loop.WaitOnRegistered()
+	require.EqualError(t, err, "Uncaught (in promise) undefined")
+}
+
+func TestEventLoopRejectString(t *testing.T) {
+	t.Parallel()
+	vu := &modulestest.VU{RuntimeField: goja.New()}
+	loop := eventloop.New(vu)
+	err := loop.Start(func() error {
+		_, err := vu.Runtime().RunString("Promise.reject('some string')")
+		return err
+	})
+	loop.WaitOnRegistered()
+	require.EqualError(t, err, "Uncaught (in promise) some string")
```

Comment:
This reads slightly wrong to me. `Uncaught (in promise)` implies that an error was thrown in the promise and not handled, not that the promise was rejected with a specific value (or `undefined` as in the above test). The tests below do make sense, though.

Reply:
This is the wording of literally every other JS running thing I have looked at: firefox, chrome, deno, node.  Everything uses this in my experience/experiments

Reply:
OK then. Still sounds confusing to me :smile: